### ISSUE TITLE
Silecing warnings in struct fields does not work for useless_conversion

### ISSUE THREAD

**Issue by eduardosm** [rust-lang/rust-clippy#5683]
Using `#[allow()]` in struct fields does not work for `useless_conversion`.

Can be checked on playground:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=5e438df7bcd3708184a12bc737fe4965

This happens on stable, beta, and nightly.

**Comment by flip1995** [comment]
```rust
S {
    #[allow(clippy::useless_conversion)]
    x: value.into()
}
```

**Comment by ebroto** [comment]
I think this is https://github.com/rust-lang/rust/issues/60681
